x/protocolpool/keeper/msg_server_test.go

```diff
@@ -337,3 +336,358 @@ func (suite *KeeperTestSuite) TestMsgClaimBudget() {
+	suite.Require().NoError(err)
+	negativePercentage, err := math.LegacyNewDecFromStr("-0.2")
+	suite.Require().NoError(err)
+	invalidExpirty := suite.ctx.BlockTime().Add(-15 * time.Second)
+	oneMonthInSeconds := int64(30 * 24 * 60 * 60) // Approximate number of seconds in 1 month
+	expiry := suite.ctx.BlockTime().Add(time.Duration(oneMonthInSeconds) * time.Second)
+	testCases := map[string]struct {
+		input     *types.MsgCreateContinuousFund
+		expErr    bool
+		expErrMsg string
+	}{
+		"empty recipient address": {
+			input: &types.MsgCreateContinuousFund{
+				Authority:  suite.poolKeeper.GetAuthority(),
+				Recipient:  "",
+				Percentage: percentage,
+				Expiry:     &expiry,
+			},
+			expErr:    true,
+			expErrMsg: "empty address string is not allowed",
+		},
+		"empty authority": {
+			input: &types.MsgCreateContinuousFund{
+				Authority:  "",
+				Recipient:  recipientAddr.String(),
+				Percentage: percentage,
+				Expiry:     &expiry,
+			},
+			expErr:    true,
+			expErrMsg: "empty address string is not allowed",
+		},
+		"invalid authority": {
+			input: &types.MsgCreateContinuousFund{
+				Authority:  "invalid_authority",
+				Recipient:  recipientAddr.String(),
+				Percentage: percentage,
+				Expiry:     &expiry,
+			},
+			expErr:    true,
+			expErrMsg: "invalid authority",
+		},
+		"zero percentage": {
+			input: &types.MsgCreateContinuousFund{
+				Authority:  suite.poolKeeper.GetAuthority(),
+				Recipient:  recipientAddr.String(),
+				Percentage: math.LegacyNewDec(0),
+				Expiry:     &expiry,
+			},
+			expErr:    true,
+			expErrMsg: "percentage cannot be zero or empty",
+		},
+		"negative percentage": {
+			input: &types.MsgCreateContinuousFund{
+				Authority:  suite.poolKeeper.GetAuthority(),
+				Recipient:  recipientAddr.String(),
+				Percentage: negativePercentage,
+				Expiry:     &expiry,
+			},
+			expErr:    true,
+			expErrMsg: "percentage cannot be negative",
+		},
+		"invalid percentage": {
+			input: &types.MsgCreateContinuousFund{
+				Authority:  suite.poolKeeper.GetAuthority(),
+				Recipient:  recipientAddr.String(),
+				Percentage: math.LegacyNewDec(1),
+				Expiry:     &expiry,
+			},
+			expErr:    true,
+			expErrMsg: "percentage cannot be greater than or equal to one",
+		},
+		"invalid expiry": {
+			input: &types.MsgCreateContinuousFund{
+				Authority:  suite.poolKeeper.GetAuthority(),
+				Recipient:  recipientAddr.String(),
+				Percentage: percentage,
+				Expiry:     &invalidExpirty,
+			},
+			expErr:    true,
+			expErrMsg: "expiry time cannot be less than the current block time",
+		},
+		"all good": {
+			input: &types.MsgCreateContinuousFund{
+				Authority:  suite.poolKeeper.GetAuthority(),
+				Recipient:  recipientAddr.String(),
+				Percentage: percentage,
+				Expiry:     &expiry,
+			},
+			expErr: false,
+		},
+	}
+
+	for name, tc := range testCases {
+		suite.Run(name, func() {
+			suite.SetupTest()
+
+			_, err := suite.msgServer.CreateContinuousFund(suite.ctx, tc.input)
```

Comment:
should the state is what is expected, i.e. recipient fund is set to 0, it can be found in the list. If todistirbute is set to a number that either the funds are distributed before creation or that todistribute still holds the previous value.

Reply:
I don't get it. How the funds are distributed before creation if toDistribute is set to a number? without withdraw request, the funds wouldn't be distributed. Even if there are other withdraw requests and creation is done after that, the funds wouldn't be distributed to it until there are funds in toDistribute and that too with withdraw request.